pallets/rmrk-core/src/functions.rs

```diff
@@ -563,6 +583,9 @@ impl<T: Config>
 
 		Nfts::<T>::remove(collection_id, nft_id);
 
+		// Remove all of the properties of the NFT
```

Comment:
Just found the `Lock` is also left after burning. Maybe there are also some other storage worth checking for clean up?

Reply:
If a NFT has a `Lock`, most cases would mean that the NFT is listed in the marketplace. This gives the NFT owner opportunity to list a NFT in the marketplace then delete it which will also leave the NFT in `ListedNfts` storage. I would add a check to prevent burning `ListedNfts` then proceed to remove properties & `Lock`. We should also remove instances where we set the `Lock` to `false` and instead just remove the `Lock`

Reply:
> I would add a check to prevent burning `ListedNfts` then proceed to remove properties & `Lock`.

But we cannot really do that inside the `rmrk-core` pallet since it doesn't depend on the `rmrk-market` pallet. So from the point of view of the `rmrk-core` the `rmrk-market` pallet doesn't even exist. 

Maybe we should not allow burning NFTs when they are locked. This way if the user has his NFT listed he will first go and unlist it and then proceed with burning.